$MirOS: ports/net/curl/patches/patch-src_main_c,v 1.1 2008/04/05 21:54:40 tg Exp $
--- src/main.c.orig	Sun Jun  1 17:58:47 2008
+++ src/main.c	Tue Aug 26 19:05:18 2008
@@ -2251,7 +2251,7 @@ static ParameterError getparameter(char 
               if(!n)
                 return PARAM_NO_MEM;
               if (nlen > 0) /* only append '=' if we have a name */
-                snprintf(n, outlen, "%.*s=%s", nlen, nextarg, enc);
+                snprintf(n, outlen, "%.*s=%s", (int)nlen, nextarg, enc);
               else
                 strcpy(n, enc);
               curl_free(enc);
@@ -2605,7 +2605,7 @@ static ParameterError getparameter(char 
               "A specified range MUST include at least one dash (-). "
               "Appending one for you!\n");
         off = curlx_strtoofft(nextarg, NULL, 10);
-        snprintf(buffer, sizeof(buffer), "%Od-", off);
+        snprintf(buffer, sizeof(buffer), "%lld-", (long long)off);
         GetStr(&config->range, buffer);
       }
       {
@@ -3373,7 +3373,7 @@ int my_trace(CURL *handle, curl_infotype
   now = localtime(&secs);  /* not multithread safe but we don't care */
   if(config->tracetime)
     snprintf(timebuf, sizeof(timebuf), "%02d:%02d:%02d.%06d ",
-             now->tm_hour, now->tm_min, now->tm_sec, tv.tv_usec);
+             now->tm_hour, now->tm_min, now->tm_sec, (int)tv.tv_usec);
   else
     timebuf[0]=0;
 
@@ -3691,7 +3691,7 @@ CURLcode _my_setopt(CURL *curl, struct C
   }
   else {
     curl_off_t oval = va_arg(arg, curl_off_t);
-    snprintf(value, sizeof(value), "(curl_off_t)%Od", oval);
+    snprintf(value, sizeof(value), "(curl_off_t)%lld", (long long)oval);
     ret = curl_easy_setopt(curl, tag, oval);
   }
 
@@ -4777,7 +4777,7 @@ operate(struct Configurable *config, int
                 /* We have written data to a output file, we truncate file
                  */
                 if(!(config->conf&CONF_MUTE))
-                  fprintf(config->errors, "Throwing away %Od bytes\n", outs.bytes);
+                  fprintf(config->errors, "Throwing away %lld bytes\n", (long long)outs.bytes);
                 fflush(outs.stream);
                 /* truncate file at the position where we started appending */
 #ifdef HAVE_FTRUNCATE
